From 74feb0759484202496e7639c5645e49a7258c9d5 Mon Sep 17 00:00:00 2001
From: Antonio Ojea <antonio.ojea.garcia@gmail.com>
Date: Wed, 12 May 2021 19:15:56 +0200
Subject: --PATCH-- kubeadm: CoreDNS permissions for endpointslices

Cherry-pick of upstream Kubernetes commit
https://github.com/kubernetes/kubernetes/commit/74feb07, which is
part of a PR (https://github.com/aws/eks-distro/pull/547) for
Kubernetes 1.22.

This change is required when using coreDNS version >=1.8.1. As stated
in the release notes:
  > "If using the kubernetes plugin for a Kubernetes cluster >= 1.19,
    CoreDNS must be granted list and watch access to endpointslices.
    (ref https://coredns.io/2021/01/20/coredns-1.8.1-release/)

Signed-off-by: Antonio Ojea <antonio.ojea.garcia@gmail.com>
Co-authored-by: pacoxu <paco.xu@daocloud.io>
---
 cmd/kubeadm/app/phases/addons/dns/manifests.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/cmd/kubeadm/app/phases/addons/dns/manifests.go b/cmd/kubeadm/app/phases/addons/dns/manifests.go
index 3ac6856bfc6..97c7f8b3e60 100644
--- a/cmd/kubeadm/app/phases/addons/dns/manifests.go
+++ b/cmd/kubeadm/app/phases/addons/dns/manifests.go
@@ -197,6 +197,13 @@ rules:
   - nodes
   verbs:
   - get
+- apiGroups:
+  - discovery.k8s.io
+  resources:
+  - endpointslices
+  verbs:
+  - list
+  - watch
 `
 	// CoreDNSClusterRoleBinding is the CoreDNS Clusterrolebinding manifest
 	CoreDNSClusterRoleBinding = `
-- 
2.25.1

